service/controller/v10/adapter/mocks.go

```diff
@@ -40,9 +46,11 @@ func (e *EC2ClientMock) DescribeSecurityGroups(input *ec2.DescribeSecurityGroups
 	return nil, fmt.Errorf("security group not found")
 }
 
-func (e *EC2ClientMock) DescribeSubnets(input *ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error) {
-	if e.subnetID == "" {
-		e.subnetID = "subnet-1234"
+func (e EC2ClientMock) DescribeSubnets(input *ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error) {
```

Comment:
Changing methods to non-pointer receiver is required to comply with ec2iface.EC2API interface.

Reply:
Otherwise we run in this issue
```
# github.com/giantswarm/aws-operator/service/controller/v10/resource/cloudformation
service/controller/v10/resource/cloudformation/create_test.go:106:6: cannot use adapter.EC2ClientMock literal (type adapter.EC2ClientMock) as type ec2iface.EC2API in field value:
        adapter.EC2ClientMock does not implement ec2iface.EC2API (DescribeSubnets method has pointer receiver)
```

Reply:
What if you pass `& adapter.EC2ClientMock`? Why `adapter` doesn't hold the pointer?

Reply:
done